controllers/dogu_controller.go

```diff
@@ -232,6 +238,50 @@ func (r *doguReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+func (r *doguReconciler) performInstallOperation(ctx context.Context, doguResource *k8sv1.Dogu) (ctrl.Result, error) {
+	installError := r.doguManager.Install(ctx, doguResource)
+	contextMessageOnError := fmt.Sprintf("failed to install dogu %s", doguResource.Name)
+
+	if installError != nil {
+		printError := strings.Replace(installError.Error(), "\n", "", -1)
+		r.recorder.Eventf(doguResource, v1.EventTypeWarning, ErrorOnInstallEventReason, "Installation failed. Reason: %s.", printError)
+	} else {
+		r.recorder.Event(doguResource, v1.EventTypeNormal, InstallEventReason, "Installation successful.")
+	}
+
+	result, handleErr := r.doguRequeueHandler.Handle(ctx, contextMessageOnError, doguResource, installError, func(dogu *k8sv1.Dogu) {
+		doguResource.Status.Status = k8sv1.DoguStatusNotInstalled
+	})
+	if handleErr != nil {
+		r.recorder.Event(doguResource, v1.EventTypeWarning, ErrorOnRequeueEventReason, "Failed to requeue the installation.")
+		return requeueWithError(fmt.Errorf(handleRequeueErrMsg, handleErr))
+	}
+
+	return requeueOrFinishOperation(result)
+}
+
+func (r *doguReconciler) performDeleteOperation(ctx context.Context, doguResource *k8sv1.Dogu) (ctrl.Result, error) {
```

Comment:
🚫 **[checkstyle]** <dupl> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>263-283 lines are duplicate of `controllers/dogu_controller.go:241-261`